xlive/Blam/Engine/shell/shell.cpp

```diff
@@ -80,107 +96,79 @@ const char* shell_get_version(void)
 
 bool shell_initialize(void)
 {
-	for (int32 i = 0; i < k_number_of_shell_command_line_flags; i++)
+	bool result = false;
+
+	if (!shell_is_dedicated_server())
 	{
-		shell_command_line_flag_set((e_shell_command_line_flags)i, 0);
+		// Don't check the result of this because we're not using GFWL
+		gfwl_gamestore_initialize();
 	}
 
-	shell_command_line_flag_set(_shell_command_line_flag_nointro, H2Config_skip_intro);
-
-	// ### TODO FIXME: voice-chat is disabled for now
-	shell_command_line_flag_set(_shell_command_line_flag_disable_voice_chat, true);
-
-	// Don't check the result of this because we're not using GFWL
-	gfwl_gamestore_initialize();
-
 	cache_file_verification_initialize();
+	cseries_initialize();
 	runtime_state_initialize();
 
-	int arg_count;
-	wchar_t** cmd_line_args = LOG_CHECK(CommandLineToArgvW(GetCommandLineW(), &arg_count));
-	if (cmd_line_args && arg_count > 1) {
-		for (int i = 1; i < arg_count; i++) {
-			wchar_t* cmd_line_arg = cmd_line_args[i];
+	SYSTEM_DEBUG_MEMORY(cseries_initialize());
```

Comment:
speaking of `SYSTEM_DEBUG_MEMORY`, the macro should also have a "before" initialization message, and all of these initialization calls should be 1 liners:

```
SYSTEM_DEBUG_MEMORY(real_math_initialize());
```

not:

```
cseries_initialize()
SYSTEM_DEBUG_MEMORY(real_math_initialize());
```

where SYSTEM_DEBUG_MEMORY is different depending on the compile configuration used

Reply:
I think this is actually how the macro was originally, since `runtime_state_initialize` is called after `cseries_initialize` but the debug memory internal call is done after `runtime_state_initialize`

Reply:
Same thing with `animation_cache_new`. There's no animation cache on halo 2 vista but a memory internal call is done after it would of been there like on xbox.

Reply:
nvm, there's no "before" message, I miscalled that

Reply:
There does seem to be a before macro as well (i.e. before XNetStartup), they're just not used in that shell code.